src/table_header_adder.rs

```diff
@@ -0,0 +1,205 @@
+use toml_edit::{Table, Item};
+use anyhow::{bail, Result};
+
+/*
+Perform an "add" at a table_header_path followed by a dotted_path.
+Example:
+table_header_path = "a/b"
+dotted_path = "c/d"
+value = "true"
+yields:
+```
+[a.b]
+c.b = true
+```
+*/
+pub fn add_value_with_table_header_and_dotted_path(
+    table: &mut Table,
+    table_header_path: &[String],
+    dotted_path: &[String],
+    value: Item) -> Result<()>
+{
+    match table_header_path.get(0) {
+        None => add_value_with_dotted_path(table, dotted_path, value),
+        Some(field) => {
+            match table.get_mut(field) {
+                Some(Item::Table(ref mut inner_table)) => {
+                    inner_table.set_dotted(table_header_path.len() > 1);
```

Comment:
what happens if you do this? i think it's fine to just always set this to dotted since `` can't be a key and `a` dotted is `a`
```suggestion
                    inner_table.set_dotted(true);
```

Reply:
The conditional is important. We want to not dot it if it's the last element of the table header path. Example: if table_header_path = "modulesConfig/languageServers", we want the languageServers element to *not* be dotted, creating:
```
[modulesConfig.languageServers]
a.b.c = d
```
if it were dotted, it would end up removing the table header, iirc.